### ISSUE TITLE
GraphQL - subjects occasionally duplicated

### ISSUE THREAD

**Issue by richardhallett** [datacite/lupo#569]
When attempting to query subject metadata, specifically fields of science, it appears to be duplicated, but in the original metadata seems fine.

```
query {
  service(id: "https://doi.org/10.25495/xkya-0g76") {
    subjects {
      subject,
      subjectScheme
    }
  }
}
```

**Comment by mfenner** [comment]
@richardhallett I think this happens when Field of Science is not provided in a specific format, where `subject` starts with `FOS:`. Can you check that this still happens with all `subject` start with `FOS:`.

**Comment by richardhallett** [comment]
I don't know what you mean, the metadata does have FOS: in it as far as I can tell.

**Comment by mfenner** [comment]
In production I updated the metadata of two service DOIs what were still using the old format. 

Just let me know if you still see the issue today, and I can look into this.

**Comment by richardhallett** [comment]
The example in the original description "10.25495/xkya-0g76" appears correct, it has this metadata:

`    <subject subjectScheme="Fields of Science and Technology (FOS)" schemeURI="http://www.oecd.org/science/inno" valueURI="http://www.oecd.org/science/inno/38235147.pdf">FOS: Computer and information sciences</subject>`

**Comment by richardhallett** [comment]
This still appears to be an issue, I registered ORCID APIs and notice that this is duplicating:
https://pidservices.org/services/10.25495/3n0t-q884

**Comment by mfenner** [comment]
Resolved by updating the metadata.

**Comment by richardhallett** [comment]
I don't see this is fixed in the general case, this still occurs on other metadata.
e.g. 10.25495/dh57-cx95 and 10.25495/912q-yc02

You said before starting with FOS: but as far as I can tell these do have this starting and is this because of the form generating bad metadata? What specifically did you change for the metadata.

**Comment by mfenner** [comment]
All DOI metadata using valueURI "http://www.oecd.org/science/inno/38235147.pdf" need to be updated.

**Comment by richardhallett** [comment]
But change it to what? As far as I can tell the subject is identical between the broken and not broken.

e.g.

Broken DOI 10.25495/dh57-cx95
```<subject subjectScheme="Fields of Science and Technology (FOS)" schemeURI="http://www.oecd.org/science/inno" valueURI="http://www.oecd.org/science/inno/38235147.pdf">FOS: Computer and information sciences</subject>```
Fixed DOI: 10.25495/3n0t-q884
```<subject subjectScheme="Fields of Science and Technology (FOS)" schemeURI="http://www.oecd.org/science/inno" valueURI="http://www.oecd.org/science/inno/38235147.pdf">FOS: Computer and information sciences</subject>```

Perhaps I'm missing something obvious.

**Comment by mfenner** [comment]
I just updated 10.25495/dh57-cx95 via the form and it now looks good to me. Can you confirm? I think valueURI is not important, but the update this triggers in the API. I think this is needed for all Service DOIs registered before the changes in the API.

**Comment by richardhallett** [comment]
Yes that seems to have fixed that DOI, but I don't understand what the difference is in the metadata, is it missing metadata? or an index problem?

I have created DOIs since presumably any API changes i.e. https://pidservices.org/services/10.25495/qq9m-c669 this one has the exact same problem and was registered last week.

So I am assuming there is something wrong with the metadata, but it is valid metadata.

Are we saying that all DOIs just need updating and then this will no longer be an issue?

**Comment by mfenner** [comment]
If you look at 10.25495/qq9m-c669 you see that it actually has two entries for FOS. Deleting one should fix it.

**Comment by richardhallett** [comment]
I can not see the two entries in the XML:

![image](https://user-images.githubusercontent.com/7294431/87286804-9beb2080-c4f9-11ea-98af-d2d6dbbdc090.png)

**Comment by mfenner** [comment]
Sorry, you see this when you open the form to edit. This is the API interfering.

**Comment by richardhallett** [comment]
So the form correctly shows the duplicate Subjects, there might be a separate issue with creating these subjects from raw XML, but that is I suspect a unrelated issue.

So the original issue of duplicated in the API can be fixed by removing the duplicates using the form for now. If we encounter this still happening, it is potentially an issue with DOI creation instead and we can make a different issue for tracking that.

Closing.
